src/core_ocean/shared/mpas_ocn_time_varying_forcing.F

```diff
@@ -0,0 +1,455 @@
+
+    if (config_use_atmospheric_forcing) then
+
+        call atmospheric_forcing(&
+             streamManager, &
+             domain, &
+             simulationClock)
+
+    endif
+
+    ! call other forcings, to be added if needed
+    ! if (config_use_****_forcing) then
+    !   call ****_forcing(&
+    !        streamManager, &
+    !        domain, &
+    !        simulationClock)
+    ! endif
+
+
+  end subroutine ocn_time_varying_forcing_get!}}}
+
+!|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
+!
+!  atmospheric_forcing
+!
+!> \brief
+!> \author Steven Brus, LANL
+!> \date 1/8/2019
+!> \details
+!>
+!
+!-----------------------------------------------------------------------
+
+  subroutine atmospheric_forcing(streamManager, domain, simulationClock)!{{{
+
+    type (MPAS_streamManager_type), intent(inout) :: streamManager
+
+    type (domain_type) :: domain
+
+    type (MPAS_clock_type) :: simulationClock
+
+    type (MPAS_time_type) :: currentForcingTime
+
+    type (block_type), pointer :: block
+
+    character(len=StrKIND), pointer :: config_dt
+
+    character(len=strKIND), pointer :: config_atmospheric_forcing_type
+
+    type (MPAS_timeInterval_type) :: timeStepESMF
+
+    real (kind=RKIND) :: dtSim
+
+    integer :: err
+
+    ! configurations
+    call mpas_pool_get_config(domain % configs, 'config_dt', config_dt)
+    call mpas_pool_get_config(domain % configs, 'config_time_varying_atmospheric_forcing_type', config_atmospheric_forcing_type)
+
+    ! convert config_dt to real
+    call mpas_set_timeInterval(timeStepESMF, timeString=config_dt,ierr=err)
+    call mpas_get_timeInterval(timeStepESMF, dt=dtSim)
+
+    ! use the forcing layer to get data
+    call MPAS_forcing_get_forcing(&
+         forcingGroupHead, &               ! forcingGroupHead
+         "ocn_atmospheric_forcing", &      ! forcingGroupName
+         streamManager, &                  ! streamManager
+         dtSim)                            ! dt
+
+    call MPAS_forcing_get_forcing_time(&
+         forcingGroupHead, &               ! forcingGroupHead
+         "ocn_atmospheric_forcing", &      ! forcingGroupName
+         currentForcingTime)               ! forcingTime
+
+
+    ! perform post forcing
+    block => domain % blocklist
+    do while (associated(block))
+
+       ! apply drag law to compute wind stress from wind speed
+       if (trim(config_atmospheric_forcing_type) == "WINDPRES") then
+          call post_atmospheric_forcing(block)
+       endif
+
+       block => block % next
+    end do
+
+  end subroutine atmospheric_forcing!}}}
+
+!|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
+!
+!  post_atmospheric_forcing
+!
+!> \brief Calculate wind stress from wind speed and apply ramp
+!> \author Steven Brus, LANL
+!> \date 1/8/2019
+!> \details Uses drag law from J.R. Garratt MWR 1977
```

Comment:
Probably better to include the full: Garratt, J.R., 1977. Review of drag coefficients over oceans and continents. Monthly weather review, 105(7), pp.915-929.